### PR TITLE
Updated Switch Transformers model card with standardized format (Issue #36979)

### PR DESCRIPTION
**Pull request by giuseppeCoccia**
# What does this PR do?

It updates the model card for Switch Transformer by following a new standardized format

#36979 

## Before submitting
- [X] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [X] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case -> [link](https://github.com/huggingface/transformers/issues/36979#issuecomment-3052596321)
- [X] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@stevhliu

### CODE REVIEW

File: docs/source/en/model_doc/switch_transformers.md [hunk 1]
```diff
@@ -14,35 +14,187 @@ rendered properly in your Markdown viewer.
...
 
-# SwitchTransformers
-
-<div class="flex flex-wrap space-x-1">
-<img alt="PyTorch" src="https://img.shields.io/badge/PyTorch-DE3412?style=flat&logo=pytorch&logoColor=white">
+<div style="float: right;">
+    <div class="flex flex-wrap space-x-1">
+        <img alt="PyTorch" src="https://img.shields.io/badge/PyTorch-2.0+-red?logo=pytorch&logoColor=white">
+        <img alt="TensorFlow" src="https://img.shields.io/badge/TensorFlow-2.0+-orange?logo=tensorflow&logoColor=white">
+        <img alt="Transformers" src="https://img.shields.io/badge/🤗%20Transformers-4.0+-blue">
```
Comment (stevhliu) [inline docs/source/en/model_doc/switch_transformers.md]: This should only have the PyTorch badge

```suggestion
```

File: docs/source/en/model_doc/switch_transformers.md [hunk 2]
```diff
@@ -14,35 +14,187 @@ rendered properly in your Markdown viewer.
...
+    </div>
 </div>
 
-## Overview
+# Switch Transformers
+
+[Switch Transformers](https://huggingface.co/papers/2101.03961) is a Mixture of Experts (MoE) model trained on Masked Language Modeling (MLM) tasks. The model architecture builds upon the classic T5 encoder-decoder framework, but replaces the traditional Feed Forward layers with **Sparse MLP layers containing "experts"** for improved efficiency and scaling.
```
Comment (stevhliu) [inline docs/source/en/model_doc/switch_transformers.md]:
```suggestion
[Switch Transformers](https://huggingface.co/papers/2101.03961) is a sparse T5 model where the MLP layer is replaced by a Mixture-of-Experts (MoE). A routing mechanism associates each token with an expert and each expert is a dense MLP. Sparsity enables better scaling and the routing mechanism allows the model to select relevant weights on the fly which increases model capacity.
```

File: docs/source/en/model_doc/switch_transformers.md [hunk 3]
```diff
@@ -14,35 +14,187 @@ rendered properly in your Markdown viewer.
...
+- **Scalable Architecture**: Enables training larger models with the same computational budget
+- **T5-Compatible**: Maintains compatibility with T5 tokenizer and generation methods
+- **Efficient Inference**: Activates only a fraction of parameters during inference
+
+Switch Transformers come in different sizes:
+
+- **switch-base-8**: 8 experts, ~223M parameters
+- **switch-base-16**: 16 experts, ~415M parameters  
+- **switch-base-32**: 32 experts, ~799M parameters
+- **switch-base-64**: 64 experts, ~1.6B parameters
```
Comment (stevhliu) [inline docs/source/en/model_doc/switch_transformers.md]: This can all be removed. Please remember to follow the format outlined in the original post!

File: docs/source/en/model_doc/switch_transformers.md [hunk 4]
```diff
@@ -14,35 +14,187 @@ rendered properly in your Markdown viewer.
...
+
+You can find all the original Switch Transformers checkpoints under the [Switch Transformers](https://huggingface.co/collections/google/switch-transformers-65f9e0b4e3f5b0af6f4dbc6c) collection.
```
Comment (stevhliu) [inline docs/source/en/model_doc/switch_transformers.md]:
```suggestion
You can find all the original Switch Transformers checkpoints under the [Switch Transformer](https://huggingface.co/collections/google/switch-transformers-release-6548c35c6507968374b56d1f) collection.
```

File: docs/source/en/model_doc/switch_transformers.md [hunk 5]
```diff
@@ -14,35 +14,187 @@ rendered properly in your Markdown viewer.
...
+
+
+> [!TIP]
+> This model was contributed by [Younes Belkada](https://huggingface.co/ybelkada) and [Arthur Zucker](https://huggingface.co/ArthurZ).
```
Comment (stevhliu) [inline docs/source/en/model_doc/switch_transformers.md]:
```suggestion
> This model was contributed by [ybelkada](https://huggingface.co/ybelkada) and [ArthurZ](https://huggingface.co/ArthurZ).
```

File: docs/source/en/model_doc/switch_transformers.md [hunk 6]
```diff
@@ -14,35 +14,187 @@ rendered properly in your Markdown viewer.
...
+>
+> Click on the Switch Transformers models in the right sidebar for more examples of how to apply Switch Transformers to different natural language tasks.
+
+The example below demonstrates how to do Masked Language Modeling with the [`Pipeline`] and the [`AutoModel`] classes, or with `transfomer-cli`.
```
Comment (stevhliu) [inline docs/source/en/model_doc/switch_transformers.md]:
```suggestion
The example below demonstrates how to predict the masked token with [`Pipeline`], [`AutoModel`], and from the command line.
```

File: docs/source/en/model_doc/switch_transformers.md [hunk 7]
```diff
@@ -14,35 +14,187 @@ rendered properly in your Markdown viewer.
...
+
+<hfoptions id="usage">
+<hfoption id="Pipeline">
+
+```python
```
Comment (stevhliu) [inline docs/source/en/model_doc/switch_transformers.md:40]:
```py
import torch
from transformers import pipeline

pipeline = pipeline(
    task="text2text-generation", 
    model="google/switch-base-8",
    torch_dtype=torch.float16,
    device=0
)
pipeline("The capital of France is <extra_id_0>.")
```

File: docs/source/en/model_doc/switch_transformers.md [hunk 8]
```diff
@@ -14,35 +14,187 @@ rendered properly in your Markdown viewer.
...
+)
+result = generator("The capital of France is <extra_id_0>.")
+print(result)
+# [{'generated_text': 'Paris.'}]
+```
+
+</hfoption>
+<hfoption id="AutoModel">
+
+```python
```
Comment (stevhliu) [inline docs/source/en/model_doc/switch_transformers.md:56]:
```py
import torch
from transformers import AutoModelForSeq2SeqLM, AutoTokenizer

tokenizer = AutoTokenizer.from_pretrained("google/switch-base-8")
model = AutoModelForSeq2SeqLM.from_pretrained("google/switch-base-8", device_map="auto", torch_dtype=torch.float16)

input_text = "The capital of France is <extra_id_0>."
input_ids = tokenizer(input_text, return_tensors="pt").input_ids.to(0)

outputs = model.generate(input_ids)
print(tokenizer.decode(outputs[0]))
```

File: docs/source/en/model_doc/switch_transformers.md [hunk 9]
```diff
@@ -14,35 +14,187 @@ rendered properly in your Markdown viewer.
...
+</hfoption>
+</hfoptions>
 
-This model was contributed by [Younes Belkada](https://huggingface.co/ybelkada) and [Arthur Zucker](https://huggingface.co/ArthurZ).
-The original code can be found [here](https://github.com/google/flaxformer/tree/main/flaxformer/architectures/moe).
+Quantization reduces the memory burden of large models by representing the weights in a lower precision. Refer to the [Quantization](../quantization/overview) overview for more available quantization backends.
 
-## Usage tips
+The example below uses [bitsandbytes](https://huggingface.co/docs/bitsandbytes/v0.46.0/index) to only quantize the weights to int8.
```
Comment (stevhliu) [inline docs/source/en/model_doc/switch_transformers.md]:
```suggestion
The example below uses [bitsandbytes](../quantization/bitsandbytes/) to only quantize the weights to 8-bits.
```

File: docs/source/en/model_doc/switch_transformers.md [hunk 10]
```diff
@@ -14,35 +14,187 @@ rendered properly in your Markdown viewer.
...
+import torch
+from transformers import AutoTokenizer, AutoModelForSeq2SeqLM, BitsAndBytesConfig
+
+tokenizer = AutoTokenizer.from_pretrained("google/switch-base-8")
+
+# Define the quantization configuration using BitsAndBytesConfig
+quantization_config = BitsAndBytesConfig(
+    load_in_8bit=True
+)
+model = AutoModelForSeq2SeqLM.from_pretrained(
```
Comment (stevhliu) [inline docs/source/en/model_doc/switch_transformers.md]: Align the rest of this code example with the one from the `AutoModel` example.

File: docs/source/en/model_doc/switch_transformers.md [hunk 11]
```diff
@@ -14,35 +14,187 @@ rendered properly in your Markdown viewer.
...
+> **Fine-tuning**: When fine-tuning Switch Transformers, be aware that the routing mechanism may need adjustment. Consider using lower learning rates for the router parameters.
+
+> [!TIP]
+> **Tokenizer Compatibility**: Switch Transformers use the same tokenizer as T5, so existing T5 preprocessing pipelines can be directly applied.
+
+> [!WARNING]
+> **Expert Utilization**: Monitor expert utilization during training to ensure balanced load distribution. Unbalanced expert usage can lead to reduced model performance.
 
-- SwitchTransformers uses the [`T5Tokenizer`], which can be loaded directly from each model's repository.
-- The released weights are pretrained on English [Masked Language Modeling](https://moon-ci-docs.huggingface.co/docs/transformers/pr_19323/en/glossary#general-terms) task, and should be finetuned.
```
Comment (stevhliu) [inline docs/source/en/model_doc/switch_transformers.md]: This can all be removed as well, remember to follow the format in the original post outlined in the issue!

File: docs/source/en/model_doc/switch_transformers.md [hunk 12]
```diff
@@ -14,35 +14,187 @@ rendered properly in your Markdown viewer.
...
 
 ## Resources
 
-- [Translation task guide](../tasks/translation)
-- [Summarization task guide](../tasks/summarization)
+- **[All Switch Transformers Checkpoints](https://huggingface.co/models?filter=switch-transformers)**: Browse available model variants
+- **[Mixture of Experts Guide](https://huggingface.co/blog/moe)**: Learn more about MoE architectures
+
```
Comment (stevhliu) [inline docs/source/en/model_doc/switch_transformers.md]:
```suggestion
```

**Review by stevhliu — COMMENTED** [review]
Good start so far!

**Comment by giuseppeCoccia** [comment]
Hi @stevhliu, thanks a lot for the detailed feedback. I have applied all the changes you suggested in a new commit.
Could you please review it again?

**Review by stevhliu — APPROVED** [review]
Thanks for your contribution! 🤗

**Comment by giuseppeCoccia** [comment]
Thank you for all that you are making at HuggingFace. I am glad I can contribute to one of the most important, if not the most important, ML libraries out there
